src/main/java/net/dv8tion/jda/internal/entities/ReceivedMessage.java

```diff
@@ -579,9 +579,6 @@ private void checkUser()
     @Override
     public AuditableRestAction<Void> delete()
     {
-        if (isEphemeral())
-            throw new IllegalStateException("Cannot delete ephemeral messages.");
```

Comment:
This still doesn't work, since this method does not use interactions.

![image](https://user-images.githubusercontent.com/18090140/197151538-e63fbde8-42ed-4f1a-bbe1-1912ce9b16b2.png)

This method uses `DELETE /channels/:channel_id/messages/:message_id`.

Reply:
Wait a sec, why would that check for if it is ephemeral even be there if you can't retrieve ephemeral messages?

Reply:
You can retrieve them

Reply:
Hmm... how do you want me to grab the interaction token then?

Reply:
Just revert the change. The only needed change is for the docs on `setEphemeral` really.

Reply:
Okay